docs/mapping.md

```diff
@@ -156,6 +156,12 @@ should not be populated at the moment. If the subscription name
 becomes available in the GCF HTTP representation, this document will
 be updated, along with the conformance tests.)
 
+Additionally, two properties should be populated in the message,
+based on the context:
+
+- The `messageId` property in the `message` object should be set to *gcf_event_id*
+- The `publishTime` proeprty in the `message` object should be set to *gcf_timestamp*
```

Comment:
Pub/Sub actually includes the snake_case variants of these properties as well `message_id` and `published_time`. See their [docs here for an example](https://cloud.google.com/pubsub/docs/push#receiving_messages). Looks like we [haven't included these in our CloudEvent schema](https://github.com/googleapis/google-cloudevents/blob/master/jsonschema/google/events/cloud/pubsub/v1/MessagePublishedData.json#L29). But if you actually deploy an CloudRun service with an EventArc trigger, these fields are included:

```
{
 "specversion": "1.0",
 "type": "google.cloud.pubsub.topic.v1.messagePublished",
 "source": "//pubsub.googleapis.com/projects/anniefu-kape-dev/topics/eventarc-us-central1-events-pubsub-trigger-935",
 "id": "2407454701990617",
 "time": "2021-05-17T19:16:43.846Z",
 "datacontenttype": "application/json",
 "data": {
   "message": {
     "data": "UnVubmVy",
     "messageId": "2407454701990617",
     "message_id": "2407454701990617",
     "publishTime": "2021-05-17T19:16:43.846Z",
     "publish_time": "2021-05-17T19:16:43.846Z"
   },
   "subscription": "projects/anniefu-kape-dev/subscriptions/eventarc-us-central1-events-pubsub-trigger-sub-935"
 }
}
```

Reply:
Yup, this was noted as part of https://github.com/googleapis/google-cloudevents/issues/151

My view is that those are effectively vestigial - they're not part of the proto describing the event (because the JSON representation of a `message_id` proto field is `messageId` etc) and I don't think we should be going out of our way to populate them. I'd like to regard the proto as canonical, and acknowledge that there may be fields not described in the proto as (these fields, and the `@type` field from the legacy event format).

Reply:
OK I am fine with this if you think it is the right thing to do. I assumed PubSub intentionally included both the `snake_case` and `camelCase` variants in order to better support a variety of language idioms, so we might want to preserve the behaviour.

Reply:
Nope, I'm pretty sure it was just accidental :)